app/integration/runtime/src/main/java/io/syndesis/integration/runtime/util/JsonSimplePredicate.java

```diff
@@ -16,101 +16,158 @@
 package io.syndesis.integration.runtime.util;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.MapLikeType;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+
 import org.apache.camel.CamelContext;
 import org.apache.camel.Exchange;
+import org.apache.camel.Message;
 import org.apache.camel.Predicate;
 import org.apache.camel.spi.Language;
 import org.apache.camel.spi.Registry;
 import org.apache.camel.util.ExchangeHelper;
 import org.apache.camel.util.ObjectHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Predicate which tries to convert a JSON message to a map first before
  * applying
  */
-public class JsonSimplePredicate implements Predicate {
+public final class JsonSimplePredicate implements Predicate {
+    private static final MapLikeType GENERIC_TYPE = TypeFactory.defaultInstance().constructMapLikeType(HashMap.class, String.class,
+        Object.class);
+
+    private static final Logger LOG = LoggerFactory.getLogger(JsonSimplePredicate.class);
+
     private static final ObjectMapper MAPPER = new ObjectMapper();
 
-    private final Predicate predicate;
+    private final ObjectMapper mapper;
+
     private final Predicate ognlPredicate;
 
-    public JsonSimplePredicate(String expression, CamelContext context) {
-        Language language = ObjectHelper.notNull(context.resolveLanguage("simple"), "simple language");
-        String ognlExpression = convertSimpleToOGNLForMaps(expression);
+    private final Predicate predicate;
+
+    public JsonSimplePredicate(final String expression, final CamelContext context) {
+        final Language language = ObjectHelper.notNull(context.resolveLanguage("simple"), "simple language");
+        final String ognlExpression = convertSimpleToOGNLForMaps(expression);
+
+        predicate = language.createPredicate(expression);
+        ognlPredicate = language.createPredicate(ognlExpression);
 
-        this.predicate = language.createPredicate(expression);
-        this.ognlPredicate = language.createPredicate(ognlExpression);
+        final Registry registry = context.getRegistry();
+        final Set<ObjectMapper> mappers = registry.findByType(ObjectMapper.class);
+
+        if (mappers.isEmpty()) {
+            mapper = MAPPER;
+        } else {
+            mapper = mappers.iterator().next();
```

Comment:
I guess we should default to MAPPER also in case of multiple ObjectMapper registered